instrumentation/opentelemetry-instrumentation-openai-v2/src/opentelemetry/instrumentation/openai_v2/patch.py

```diff
@@ -144,27 +132,63 @@ def _set_response_attributes(span, result):
         )
 
 
+class ToolCallBuffer:
+    def __init__(self, index, tool_call_id, function_name):
+        self.index = index
+        self.function_name = function_name
+        self.tool_call_id = tool_call_id
+        self.arguments = []
+
+    def append_arguments(self, arguments):
+        self.arguments.append(arguments)
+
+
+class ChoiceBuffer:
+    def __init__(self, index):
+        self.index = index
+        self.finish_reason = None
+        self.text_content = []
+        self.tool_calls_buffers = []
+
+    def append_text_content(self, content):
+        self.text_content.append(content)
+
+    def append_tool_call(self, tool_call):
+        idx = tool_call.index
+        # make sure we have enough tool call buffers
+        for i in range(len(self.tool_calls_buffers), idx + 1):
+            self.tool_calls_buffers.append(None)
+
+        if not self.tool_calls_buffers[idx]:
+            self.tool_calls_buffers[idx] = ToolCallBuffer(
+                idx, tool_call.id, tool_call.function.name
+            )
+        self.tool_calls_buffers[idx].append_arguments(
+            tool_call.function.arguments
+        )
+
+
 class StreamWrapper:
     span: Span
-    response_id: str = ""
```

Comment:
I'm not too familiar with the concept of `Stream`. How does it conceptually map to a `Choice` or number of `Choice`s?

Reply:
You can ask model to return a completion (one or more choices) using regular HTTP response body or you can ask it to stream tokens one-by-one - https://platform.openai.com/docs/api-reference/making-requests - in which case model returns the HTTP response and then writes tokens to the response stream using  SSE events. Check out  test recordings to see what happens there with and without streaming.

I.e. streaming affects how the response is received, but not the contents (choices).